src/bone.js

```diff
@@ -846,9 +847,13 @@ class Bone {
     const data = {};
     const Model = this;
     const { attributes } = Model;
-    for (const name in values) {
-      if (this.hasAttribute(name)) {
-        data[name] = values[name];
+    for (const key in attributes) {
+      const attribute = attributes[key];
+      if (attribute.primaryKey) continue;
+      if (!values[key] && attribute.defaultValue != null) {
```

Comment:
How about values such as `false`, `0`, or `''`? I think `values[key] == null` should be preferred.